modules/opensearch/opensearch.client.inc

```diff
@@ -532,6 +558,40 @@ function _opensearch_cache_insert(TingClientRequest $request, $response, array &
   }
 }
 
+/**
+ * Retrieve a earch response based on a request.
+ *
+ * @param TingClientRequest $request
+ *   The search request for which to retrieve a cached response.
+ * @param array $opensearch_static_fast
+ *   The fast static cache used by opensearch_execute().
+ *
+ * @return \TingClientSearchResult|\TingClientObjectCollection|\TingClientObject[]|NULL
+ *   The cached response. NULL if no cached entry exists.
+ */
+function _opensearch_cache_get(TingClientRequest $request, array &$opensearch_static_fast) {
+  // Build cache id for this request.
+  $cid = _opensearch_cache_id($request);
+  if (array_key_exists($cid, $opensearch_static_fast)) {
+    return $opensearch_static_fast[$cid];
+  }
+
+  // Check the database/memcached cache for the request also check that the
+  // cache data has not expired before using it.
+  $cache = cache_get($cid, 'cache_opensearch');
+  if ($cache && ($cache->expire > REQUEST_TIME)) {
+    $data = $cache->data;
+
+    // Update the static cache with the database cache content to speed up the
+    // next request for the same data.
+    $opensearch_static_fast[$cid] = $data;
+
+    return $data;
+  }
+
+  return NULL;
```

Comment:
Note: in the old cache we hade a special value for "empty cache" as "null" would be cached. I don't known if this an issue in the new cache...

Reply:
Has I remember it was in relation to "0-hits" searches.

Reply:
I do not think the suggested change makes the situation any better or worse.